### PR TITLE
Adding Sphinx documentation generation for the docs.

### PR DESCRIPTION
**Pull request by mithro**
Signed-off-by: Tim 'mithro' Ansell <tansell@google.com>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+dmypy.json
+
+# Pyre type checker
+.pyre/
+
+# Emacs temporary files
+*~
+
+# Generated files
+/fasm/parser/tags.py
```
Comment (gatecat) [inline .gitignore:123]: I think these gitignore entries are probably from another repo?

File: docs/Makefile [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+
+SHELL = /bin/bash
+MAKEDIR := $(dir $(lastword $(MAKEFILE_LIST)))
+
+# You can set these variables from the command line.
+SPHINXOPTS      =
+SPHINXBUILD     = [ -e env/bin/activate ] && source env/bin/activate; sphinx-build
+SPHINXAUTOBUILD = [ -e env/bin/activate ] && source env/bin/activate; sphinx-autobuild
+SPHINXPROJ      = SymbiFlowV2X
```
Comment (gatecat) [inline docs/Makefile:18]: this again looks like something from another repo

File: docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+# (source start file, target name, title, author,
+#  dir menu entry, description, category)
+texinfo_documents = [
+    (
+        master_doc,
+        'SymbiFlowFIFM',
+        u'SymbiFlow FPGA Interchange Format Documentation',
+        author,
+        'SymbiFlowFIF',
+        'One line description of project.',
```
Comment (gatecat) [inline docs/conf.py:258]: should this be filled out?

Comment (mithro) [reply docs/conf.py:258]: Let's do this later.

File: .gitignore (hunk 1 above)
Comment (mithro) [reply .gitignore]: Fixed.

File: docs/Makefile (hunk 2 above)
Comment (mithro) [reply docs/Makefile]: Fixed.

**Review by gatecat — APPROVED** [review]

**Pull request merged**